.github/workflows/test.yml

```diff
@@ -0,0 +1,310 @@
+
+      - name: Smoke test scripts
+        if: matrix.npm == 6
+        uses: "./.github/actions/smoketestscripts"
+
+      - name: Create MRT credentials file
+        if: github.repository == 'SalesforceCommerceCloud/pwa-kit' && matrix.npm == 6
+        env:
+          MOBIFY_CLIENT_USER: ${{ secrets.MOBIFY_CLIENT_USER }}
+          MOBIFY_CLIENT_API_KEY: ${{ secrets.MOBIFY_CLIENT_API_KEY }}
+        run: |-
+          # Add credentials file at ~/.mobify so we can upload to Mobify Cloud
+          npm run save-credentials --prefix packages/template-retail-react-app -- --user "$MOBIFY_CLIENT_USER" --key "$MOBIFY_CLIENT_API_KEY"
+        shell: bash
+
+      - name: Push Bundle to MRT
+        if: github.repository == 'SalesforceCommerceCloud/pwa-kit' && matrix.npm == 6 
+        #TODO: Change target 'test-env' to 'staging' and 'test-env-3' to 'commerce-sdk-react'
+        run: |-
+          if [[ $DEVELOP || $RELEASE ]]; then
+            target=test-env
+            target_commerce_sdk_react="test-env-3"
+          elif [[ $RELEASE ]]; then
+            target=production
+          else
+            target=""
+          fi
+          project="scaffold-pwa"
+          if [[ $target ]]; then
+            npm run push --prefix packages/template-retail-react-app -- -s $project --message "build ${{ github.run_id }} on ${{ github.ref }} (${{ github.sha }})" --target $target
+          fi
+          if [[ $target_commerce_sdk_react ]]; then
+            npm run push --prefix packages/test-commerce-sdk-react -- -s $project --message "test-commerce-sdk-react build ${{ github.run_id }} on ${{ github.ref }} (${{ github.sha }})" --target $target_commerce_sdk_react
+          fi
+      #TODO: remove --dry-run from install command
+      - name: Publish to NPM
+        if: github.repository == 'SalesforceCommerceCloud/pwa-kit' && matrix.npm == 6
+        env: 
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: |-
+          # Add NPM token to allow publishing
+          echo "//registry.npmjs.org/:_authToken=$NODE_AUTH_TOKEN" > ~/.npmrc
+
+          if [[ $RELEASE ]]; then
+            # Publish all changed packages. The "from-package" arg means "look
+            # at the version numbers in each package.json file and if that doesn't
+            # exist on NPM, publish"
+            npm run lerna -- publish from-package --yes --no-verify-access --dry-run
+          fi
+
+          # Cleanup
+          rm ~/.npmrc
+      - name: Send GitHub Action data to Slack workflow (PWA Kit)
+        id: slack
+        if: github.repository == 'SalesforceCommerceCloud/pwa-kit' && failure() && github.ref == 'refs/heads/develop' && matrix.npm == 6
+        uses: slackapi/slack-github-action@v1.23.0
+        with:
+          payload: |
+            {
+              "test": "testNode${{ matrix.node }}"
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+  pwa-kit-windows:
```

Comment:
The [docs](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy) shows that os is a parameter for matrix, could we combine the workflows `pwa-kit` and `pwa-kit-windows`?

something like:
```
pwa-kit:
    strategy:
        matrix:
            node: [14]
            npm: [6, 7, 8]
            os: [ubuntu, windows]
```

Reply:
Yes we can! It was initially like this but I thought it would be better to separate ubuntu and windows as there were a lot of steps that were checking for the conditional `matrix.os == 'ubuntu-latest'`